Products/GenericSetup/tool.py

```diff
@@ -921,17 +921,30 @@ def getProfileInfo(self, profile_id):
 
     @security.protected(ManagePortal)
     def getDependenciesForProfile(self, profile_id):
-        if profile_id is None:
-            return ()
-        if profile_id.startswith("snapshot-"):
-            return ()
 
-        if not self.profileExists(profile_id):
-            raise KeyError(profile_id)
```

Comment:
I can't see why raising for a non-existent profile is wrong, either.

Reply:
Not wrong, just superfluous, because instead in line 931 `getProfileInfo` will already fail with a KeyError, which also contains the profile-id, nothing more to do here, AFAICS.